mig/wsgi-bin/migwsgi.py

```diff
@@ -353,6 +354,11 @@ def application(environ, start_response):
         start_entry['headers'] = default_headers
     response_headers = start_entry['headers']
 
+    if ret_code == SIZE_LIMIT_ERROR[0]:
```

Comment:
Is the status code as returned in the case of an error being ignored currently? This seems entirely too specific to a single type of error - at minimum, if we re going down this road, we'd surely want _anything_ in the 400 range to be set as the output status code.

On a larger note, I'm a little uneasy that placing the limit inside the cat functionality file was to avoid changes to the migwsgi but we now seem to be making them anyway. Again, is this in fact fixing a pre-existing bug?

Reply:
Let me try to be a little more helpful here so I will suggest an alternative change that will avoid setting an explicit 'Status' header rather than passing what we want straight into `starts_response()`:
At this point in the wsgi wrapper we have the output objects and the ret_val tuple:
https://github.com/ucphhpc/migrid-sync/blob/edge/mig/wsgi-bin/migwsgi.py#L321
we won't throw an exception here, so I presume we are falling through with status set to '200 OK'. I think this might be an even better place to do something like:
```
if _is_error_ret_val(ret_val):
    status = ' '.join(ret_val)
else:
    status = '200 OK'
```
where we define a function in the file like:
```
def _is_error_ret_val(status):
    status_code = int(status[0])
    return status_code >= 400 status_code <= 599 
```

Doing it there means that status will be passed correctly into start_response() and thus fiddling with headers would not be needed.

Reply:
Thanks for the constructive feedback :+1: 
There is the issue that CGI appears to lack a similar structure and requires explicit headers for it, which threw me off that path earlier today ... but I'll take another look with fresh eyes next week.

Reply:
I've reverted to use 422 as suggested but adjusted the variable name and associated HTTP status text to 'Unprocessable Content' as used elsewhere.
We should look into a more general error code handling as you suggested in a separate follow-up task if we want to improve that area.